pkg/k8sutils/job_test.go

```diff
@@ -740,6 +740,88 @@ func TestAwaitK8sJobDoneErrorJobFailed(t *testing.T) {
 	)
 }
 
+func TestAwaitK8sJobDoneNotStarted(t *testing.T) {
+	k8sClientSet := k8sfake.NewSimpleClientset()
+	k8s := K8sImpl{clientset: k8sClientSet}
+
+	jobName := "job-not-started"
+
+	job := v1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: jobName,
+		},
+		Spec: v1.JobSpec{},
+		Status: v1.JobStatus{
+			Conditions: []v1.JobCondition{}, // this job won't have a condition set
+		},
+	}
+	namespace := "job-pain-and-misery-ns"
+	k8sClientSet.BatchV1().Jobs(namespace).Create(
+		context.Background(), &job, metav1.CreateOptions{},
+	)
+
+	err := k8s.AwaitK8sJobDone(jobName, 1*time.Second, 50*time.Millisecond, namespace)
+
+	require.Error(t, err)
+
+	assert.ErrorContains(
+		t, err, "max poll count reached for job",
+	)
+}
+
+func TestGetFailedEventsForJob(t *testing.T) {
+	k8sClientSet := k8sfake.NewSimpleClientset()
+	k8s := K8sImpl{clientset: k8sClientSet}
+
+	jobName := "happy-path-job"
+	namespace := "happy-path-ns"
+	failedEventMessage1 := "Failed to create pod"
+	failedEventMessage2 := "Failed to pull image"
+	successfulEventMessage1 := "Successfully pulled image"
+
+	// job config
+	job := v1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: jobName,
+		},
+		Spec: v1.JobSpec{},
+		Status: v1.JobStatus{
+			Conditions: []v1.JobCondition{}, // this job won't have a condition set
```

Comment:
I'm not sure about the Kubernetes API at this point, but shouldn't the JobStatus be completely empty if the job hangs in at the point of not being able to create a single pod?

Reply:
Using it the same way as in other tests - I believe that's the correct approach